### PR TITLE
[DAGCombiner] Avoid the pre-truncate of BUILD_VECTOR sources.

### PR DESCRIPTION
**Pull request by Asher8118**
Avoid the pre-truncate of BUILD_VECTOR sources when there is more than one use. This can avoid using unnecessary movs later down the instruction selection pipeline.

### CODE REVIEW

File: llvm/test/CodeGen/AArch64/avoid-pre-trunc.ll [hunk 1]
```diff
@@ -0,0 +1,89 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 4
+; RUN: llc -mtriple=aarch64 < %s | FileCheck %s
+
+define i32 @lower_lshr(<4 x i32> %a, <4 x i32> %b, <4 x i32> %c, <4 x i32> %d, <4 x i32> %e, <4 x i32> %f, <4 x i32> %g, <4 x i32> %h) local_unnamed_addr #0 {
```
Comment (arsenm) [inline llvm/test/CodeGen/AArch64/avoid-pre-trunc.ll:4]: Don't need local_unnamed_addr and the attributes

Comment (arsenm) [inline llvm/test/CodeGen/AArch64/avoid-pre-trunc.ll:4]: can you pre-commit the test to show the diff? A few more vector types also wouldn't hurt

Comment (davemgreen) [reply llvm/test/CodeGen/AArch64/avoid-pre-trunc.ll:4]: +1 to this :)

Comment (Rin18) [reply llvm/test/CodeGen/AArch64/avoid-pre-trunc.ll]: Added the pre-commit and some more vector types.

Comment (Rin18) [reply llvm/test/CodeGen/AArch64/avoid-pre-trunc.ll]: My bad for not spotting it, I've removed it now.

**Review by RKSimon — APPROVED** [review]
LGTM - cheers

**Pull request merged**